webapp-django/crashstats/signature/views.py

```diff
@@ -424,3 +424,112 @@ def signature_graph_data(request, channel):
     )
 
     return data
+
+
+@waffle_switch('signature-report')
+def signature_summary(request):
+    '''Return a list of specific aggregations. '''
+    params = get_validated_params(request)
+    if isinstance(params, http.HttpResponseBadRequest):
+        # There was an error in the form, let's return it.
+        return params
+
+    signature = params['signature'][0]
+
+    data = {}
+
+    current_query = request.GET.copy()
+    if 'page' in current_query:
+        del current_query['page']
+
+    data['params'] = current_query.copy()
+
+    params['signature'] = '=' + signature
+    params['_results_number'] = 0
+    params['_facets'] = [
+        'platform',
+        'cpu_name',
+        'process_type',
+        'flash_version',
+    ]
+    params['_histogram.uptime'] = ['product']
+    params['_histogram_interval.uptime'] = 60
+
+    # If the user has permissions, show exploitability.
+    all_fields = SuperSearchFields().get()
+    if has_permissions(
+        request.user, all_fields['exploitability']['permissions_needed']
+    ):
+        params['_histogram.date'] = ['exploitability']
+
+    api = SuperSearchUnredacted()
+    try:
+        search_results = api.get(**params)
+    except models.BadStatusCodeError, e:
+        # We need to return the error message in some HTML form for jQuery to
+        # pick it up.
+        return http.HttpResponseBadRequest('<ul><li>%s</li></ul>' % e)
```

Comment:
Seriously, that's copy/pasting from what we do in all other Super Search-like views. It gets sent back as AJAX, so a user will never see an actual 400 page.